### PR TITLE
Feat/msdk 121 graphql types

### PR DESCRIPTION
**Pull request by shuffledex**

### CODE REVIEW

File: scripts/fetchHarvesterTypes.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/* eslint-disable */
+const git = require('nodegit');
+const path = require('path');
+const rimraf = require('rimraf');
+const { exec } = require('child_process');
+
+const BRANCH = process.env.BRANCH;
+const GITHUB_PROYECT = 'git@github.com:PolymathNetwork/tooling-gql.git';
```
Comment (monitz87) [inline scripts/fetchHarvesterTypes.js:8]: GITHUB_PROJECT

File: scripts/fetchHarvesterTypes.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/* eslint-disable */
+const git = require('nodegit');
+const path = require('path');
+const rimraf = require('rimraf');
+const { exec } = require('child_process');
+
+const BRANCH = process.env.BRANCH;
```
Comment (monitz87) [inline scripts/fetchHarvesterTypes.js:7]: This should just be a parameter of the script

File: scripts/fetchHarvesterTypes.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+const GITHUB_PROYECT = 'git@github.com:PolymathNetwork/tooling-gql.git';
+const TOOLING_FOLDER = 'tooling-gql';
+const OUTPUT_TYPES = path.resolve(__dirname, '../src/polkadot/harvester-types.ts');
```
Comment (monitz87) [inline scripts/fetchHarvesterTypes.js:10]: This has nothing to do with polkadot, it should go into its own folder

`src/harvester/types.ts`

File: scripts/fetchHarvesterTypes.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    fetchOpts: {
+      callbacks: {
+        certificateCheck: () => 1,
+        credentials: function(_, username) {
+          return git.Cred.sshKeyFromAgent(username);
+        },
+      },
+    },
+  })
+  .then(function() {
```
Comment (monitz87) [inline scripts/fetchHarvesterTypes.js:26]: You should use nodegit itself to checkout to the branch and fetch the file, and fs to write it, not exec and unix specific commands (not platform agnostic)

File: scripts/fetchHarvesterTypes.js (hunk 2 above)
Comment (shuffledex) [reply scripts/fetchHarvesterTypes.js:7]: I dont understand

Comment (shuffledex) [reply scripts/fetchHarvesterTypes.js:7]: do you mean set it as variable in the code?

Comment (monitz87) [reply scripts/fetchHarvesterTypes.js:7]: No, I mean receive it as a parameter, like `yarn my-script some-branch`

File: .eslintrc [hunk 5]
```diff
@@ -33,5 +33,5 @@
     "project": "./tsconfig.json"
   },
   "extends": ["standard", "semistandard", "plugin:@typescript-eslint/recommended", "prettier"],
-  "ignorePatterns": ["src/polkadot/"]
+  "ignorePatterns": ["src/polkadot/", "src/harvester"]
```
Comment (monitz87) [inline .eslintrc:36]:
```suggestion
  "ignorePatterns": ["src/polkadot/", "src/harvester/"]
```

**Pull request merged**